cpp/src/arrow/record_batch.cc

```diff
@@ -167,30 +188,41 @@ class SimpleRecordBatch : public RecordBatch {
     return RecordBatch::Validate();
   }
 
+  std::shared_ptr<Device::SyncEvent> GetSyncEvent() const override { return sync_event_; }
```

Comment:
Are there any override that would have to create one `Device::SyncEvent` per call? If not, this should return a `const ...&` to avoid the non-trivial cost of incrementing a ref-count on every call.

Reply:
In theory I can't think of one, but that doesn't mean there *can't* be a situation like that.

But that's still a good point. I'd rather default to having it return `const ...&` for now, and update this in the future if necessary.